src/main/java/uk/gov/companieshouse/controller/AppealController.java

```diff
@@ -0,0 +1,110 @@
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+import uk.gov.companieshouse.model.Appeal;
+import uk.gov.companieshouse.service.AppealService;
+
+import javax.validation.Valid;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+@Slf4j
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/companies")
+public class AppealController {
+
+    public final AppealService appealService;
+
+    @Operation(summary = "Create a new appeal", tags = "Appeal")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "201", description = "Appeal resource created", headers = {
+            @Header(name = "location")
+        }),
+        @ApiResponse(responseCode = "401", description = "Unauthorised request"),
+        @ApiResponse(responseCode = "422", description = "Invalid appeal data"),
+        @ApiResponse(responseCode = "500", description = "Internal server error")
+    })
+    @PostMapping(value = "/{company-id}/appeals", consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<String> submitAppeal(@RequestHeader("ERIC-identity") String userId,
+                                               @PathVariable("company-id") final String companyId,
+                                               @Valid @RequestBody final Appeal appeal) {
+
+        log.info("POST /companies/{}/appeals with user id {} and appeal data {}",
+            companyId, userId, Json.pretty(appeal));
+
+        if (StringUtils.isBlank(userId)) {
+            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
+        }
+
+        try {
+            String id = appealService.saveAppeal(companyId, appeal, userId);
+
+            URI location = ServletUriComponentsBuilder
+                .fromCurrentRequest()
+                .path("/{id}")
+                .buildAndExpand(id)
+                .toUri();
+
+            return ResponseEntity.created(location).build();
+
+        } catch (Exception ex) {
+            log.error("Unable to create appeal for company number {} and user id {}", companyId, userId, ex);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+        }
+    }
+
+    @Operation(summary = "Get an appeal by ID", tags = "Appeal")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "Appeal resource retrieved successfully"),
+        @ApiResponse(responseCode = "404", description = "Appeal not found")
```

Comment:
No 500 mentioned in doc but I am sure it can happen.

Reply:
Added 500 response code